### PR TITLE
Sign stats request

### LINKED ISSUE #115: Expand stats endpoint with signature check
`/api/v1/stats` endpoint should be expanded with a check for signature inside the body of the request so that new `Payout` can be saved when this endpoint is invoked with the appropriate signature

### PR DESCRIPTION
**Pull request by MakMuftic**
**Short description of work done**

### PR Checklist

- [x] I have run linter localy
- [x] I have run unit and integration tests locally
- [x] Rebased to master branch / merged master
- [x] Updated CHANGELOG.md

### Changes

- Added handler for `POST /api/v1/stats`
- Refactor code for invoking stats endpoint on payout

### Example

Now I exposed two different things on **stats endpoint**:

`GET /api/v1/stats` - returns statistics for all nodes

`POST /api/v1/stats` + `X-Signature` header with valid signature - returns statistics + fee and starts new payout

### Issues

Closes #115

### CODE REVIEW

File: internal/configuration/configuration.go [hunk 1]
```diff
@@ -14,6 +14,7 @@ type Configuration struct {
 	Port                int32
 	PortPool            server.Pooler
 	TunnelServerAddress string
+	PrivateKey          string
```
Comment (MakMuftic) [inline internal/configuration/configuration.go]: ~@mpetrun5 @mpetrunic I am not sure if this is the best place to hold the private key. I could embed it as the private property of `ApiController`. I need to save it somehow so I could verify signatures in stats handler.~

Comment (MakMuftic) [reply internal/configuration/configuration.go]: I refactored this

File: internal/script/payout.go [hunk 2]
```diff
@@ -31,17 +34,28 @@ func ExecutePayout(secret string, totalReward float64, loadbalancerUrl *url.URL)
 	)
 }
 
-func fetchStatsFromEndpoint(endpoint *url.URL) (*controllers.StatsResponse, error) {
-	resp, err := http.Get(endpoint.String())
+func fetchStatsFromEndpoint(endpoint *url.URL, secret string) (*controllers.LoadbalancerStatsResponse, error) {
```
Comment (mpetrunic) [inline internal/script/payout.go:37]: privateKey

File: internal/controllers/signaturestats.go [hunk 3]
```diff
@@ -0,0 +1,5 @@
+package controllers
+
+func GetStatsSignedData() string {
+	return "loadbalancer-request"
```
Comment (mpetrun5) [inline internal/controllers/signaturestats.go]: What does this do?

Comment (MakMuftic) [reply internal/controllers/signaturestats.go]: This is a constant that represents data being signed on stats request (as there is no payload), I just wrapped it as a function.

Comment (MakMuftic) [reply internal/controllers/signaturestats.go]: @mpetrun5 do you think that it is better just to hold it as constant?

Comment (mpetrun5) [reply internal/controllers/signaturestats.go]: This is a bit counter intuitive

Comment (MakMuftic) [reply internal/controllers/signaturestats.go]: I refactored it to const

**Review by mpetrun5 — APPROVED** [review]

**Pull request merged**